crates/paimon/src/spec/data_file.rs

```diff
@@ -90,16 +72,173 @@ pub struct DataFileMeta {
     pub delete_row_count: Option<i64>,
     // file index filter bytes, if it is small, store in data file meta
     pub embedded_index: Option<Vec<u8>>,
-    pub file_source: Option<FileSource>,
 }
 
 impl Display for DataFileMeta {
-    fn fmt(&self, _: &mut Formatter<'_>) -> std::fmt::Result {
-        todo!()
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
```

Comment:
`Debug` will generate this automatically; we don't need to implement it manually.